pandas/core/reshape.py

```diff
@@ -182,9 +185,21 @@ def get_new_values(self):
         stride = values.shape[1]
         result_width = width * stride
         result_shape = (length, result_width)
+        mask = self.mask
+        mask_all = mask.all()
```

Comment:
@jreback couldn use some help grokking how we get here.  In Block._unstack adding an assertion `assert mask.all()` doesn't break any tests.  is that something we can rely on?  (if so we can simplify code a good bit)  If not, how can we construct a counter-example?